core/src/main/java/org/web3j/protocol/core/filters/Filter.java

```diff
@@ -110,7 +118,14 @@ private void pollFilter(EthFilter ethFilter) {
             throwException(e);
         }
         if (ethLog.hasError()) {
-            throwException(ethLog.getError());
+            Error error = ethLog.getError();
+            switch (error.getCode()) {
+                case RpcErrors.FILTER_NOT_FOUND: reinstallFilter();
+                    break;
+                default:
+                    break;
+            }
+            throwException(error);
```

Comment:
We shouldn't be throwing the exception if we're reinstalling the filter.

It should log a warning that the filter has not been found indicating a possible client disconnect.

Reply:
https://github.com/web3j/web3j/pull/609